modules/web/js/ballerina/diagram/views/default/components/transform/transform-node-manager.js

```diff
@@ -48,10 +48,24 @@ class TransformNodeManager {
      */
     getCompatibility(source, target) {
         const compatibility = this._typeLattice.getCompatibility(source, target);
-        if (!compatibility || (source === target)) {
+        if (source === target) {
             return { safe: true, type: 'implicit' };
         }
-        return compatibility;
+
+        // If the source and target are builtin types, we consider the mapping illegal
+        // if it is not available in the type lattice.
+        const builtInTypes = this._environment.getTypes();
+        if (!compatibility && builtInTypes.includes(source) && builtInTypes.includes(target)) {
+            return undefined;
+        }
+        return { safe: true, type: 'implicit' };
+    }
+
+    isConnectionValid(source, target) {
```

Comment:
This function is not used? I think because `!compatibility` works and looks good we don't this.

Reply:
ah. never mind. saw the usage.